kin-ecosystem-sdk/src/test/java/com/kin/ecosystem/data/order/OrderRepositoryTest.java

```diff
@@ -330,6 +339,54 @@ public void onFailure(Throwable t) {
         verify(offerRepository).setPendingOfferByID(offerID);
     }
 
+    @Test
+    public void purchase_Conflict_GetOrderConfirmation() throws Exception {
+        final CountDownLatch countDownLatch = new CountDownLatch(1);
+
+        Order confirmedOrder = new Order().orderId(orderID).offerId(offerID).status(Status.COMPLETED);
+        confirmedOrder.setResult(
+            new JWTBodyPaymentConfirmationResult().jwt("A JWT CONFIRMATION").type(TypeEnum.PAYMENT_CONFIRMATION));
+        ObservableData<Offer> pendingOffer = ObservableData.create(offer);
+
+        // Create the Conflict error response
+        Error error = Mockito.mock(Error.class);
+        when(error.getCode()).thenReturn(4091);
+
+        Map<String, List<String>> responseHeaders = new HashMap<>();
+        responseHeaders.put("location", Arrays.asList("v/some/OrderID12"));
+
+        ApiException apiException = Mockito.mock(ApiException.class);
+        when(apiException.getCode()).thenReturn(409);
+        when(apiException.getResponseBody()).thenReturn(error);
+        when(apiException.getResponseHeaders()).thenReturn(responseHeaders);
+
+        when(remote.createExternalOrderSync(anyString())).thenThrow(apiException);
+        when(remote.getOrderSync(anyString())).thenReturn(confirmedOrder);
+        when(offerRepository.getPendingOffer()).thenReturn(pendingOffer);
+
+        // Check not error, and got jwt confirmation
+        orderRepository.purchase("A GENERATED NATIVE OFFER JWT", new Callback<OrderConfirmation>() {
+            @Override
+            public void onResponse(OrderConfirmation orderConfirmation) {
+                countDownLatch.countDown();
+                assertEquals("A JWT CONFIRMATION", orderConfirmation.getJwtConfirmation());
```

Comment:
If I recall correctly there might be a problem asserting not in the main (tests) thread

Reply:
this is why we have the countDownLatch